src/mscorlib/src/System/Environment.cs

```diff
@@ -1312,52 +1313,52 @@ internal static String GetStackTrace(Exception e, bool needFileInfo)
         // Using the params object[] overload works against this since the
         // initialization of the array is done inline in the caller at the IL
         // level. So we have overloads that simply wrap the params one, and
-        // are tagged as NoInlining. That way they do not bloat either the
-        // IL or the generated asm.
+        // the methods they call through to are tagged as NoInlining. 
+        // In mscorlib NoInlining causes the caller and callee to not be inlined
+        // as it is an assumption of StackCrawlMark use so it is not added 
+        // directly to these methods, but to the ones they call.
+        // That way they do not bloat either the IL or the generated asm.
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
         internal static string GetResourceString(string key, object val0)
         {
             return GetResourceString(key, new object[] { val0 });
         }
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
         internal static string GetResourceString(string key, object val0, object val1)
         {
             return GetResourceString(key, new object[] { val0, val1 });
         }
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
         internal static string GetResourceString(string key, object val0, object val1, object val2)
         {
             return GetResourceString(key, new object[] { val0, val1, val2 });
         }
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
         internal static string GetResourceString(string key, object val0, object val1, object val2, object val3)
         {
             return GetResourceString(key, new object[] { val0, val1, val2, val3 });
         }
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
         internal static string GetResourceString(string key, object val0, object val1, object val2, object val3, object val4)
         {
             return GetResourceString(key, new object[] { val0, val1, val2, val3, val4 });
         }
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
         internal static string GetResourceString(string key, object val0, object val1, object val2, object val3, object val4, object val5)
         {
             return GetResourceString(key, new object[] { val0, val1, val2, val3, val4, val5 });
         }
 
-        [System.Security.SecuritySafeCritical]  // auto-generated
-        internal static String GetResourceString(String key, params Object[] values) {
-            String s = GetResourceString(key);
-            return String.Format(CultureInfo.CurrentCulture, s, values);
+
+        // NoInlining causes the caller and callee to not be inlined in mscorlib as it is an assumption of StackCrawlMark use
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static String GetResourceString(string key, params object[] values)
```

Comment:
This should be internal, otherwise things like

```cs
Environment.GetResourceString("Foo", new object[] { ... });
```

will break since they now call the `(string, object)` overload.